Packs/NetQuestOMX/Integrations/NetQuestOMX/NetQuestOMX.py

```diff
@@ -0,0 +1,327 @@
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy)
+        self.credentials = {"UserName": credentials["identifier"], "Password": credentials["password"]}
+        self.headers = {"Accept": "application/json"}
+        self.login()
+
+    def login(self):
+        """
+        In this method, the validity of the Access Token is checked, since the Access Token has a 120 minutes validity period.
+        Refreshes the token as needed.
+        """
+        now = datetime.utcnow()
+
+        if (cache := get_integration_context()) and cache.get("Token"):
+            expiration_time = datetime.strptime(
+                cache["expiration_time"], DATE_FORMAT_FOR_TOKEN
+            )
+
+            # check if token is still valid, continue using it. otherwise regenerate a new one
+            if (seconds_left := (expiration_time - now).total_seconds()) > 0:
+                demisto.debug(f"No need to regenerate the token, it is still valid for {seconds_left} more seconds")
+                return
+
+        demisto.debug("IntegrationContext token cache is empty or token has expired, regenerating a new token")
+        raw_response = self._refresh_access_token()
+        self._set_headers(raw_response)
+
+        set_integration_context(
+            {
+                "Token": self.headers["X-Auth-Token"],
+                "expiration_time": (
+                    now + timedelta(seconds=(TOKEN_TTL - 60))  # decreasing 60s from token expiry for safety
+                ).strftime(DATE_FORMAT_FOR_TOKEN),
+            }
+        )
+
+    def _refresh_access_token(self) -> dict:
+        """
+        Since the validity of the Access Token is 120 minutes, this method refreshes it and returns the new token json.
+
+        """
+
+        try:
+            response_json = self._http_request(
+                method="POST", url_suffix="/api/SessionService/Sessions", headers=self.headers, data=self.credentials
+            )
+        except Exception as e:
+            raise DemistoException(
+                "An error was occurred when creating a new token. Please verify your credentials."
+            ) from e
+        return response_json
+
+    def _set_headers(self, raw_response: dict):
+        """
+        This method is called during the client's building or when a new token is generated since the old one has expired.
+        """
+        # TODO - Ensure the access
+        self._headers["X-Auth-Token"] = raw_response["Headers"]["X-Auth-Token"]
+
+    def address_list_upload_request(self, file_name: str) -> requests.Response:
+        try:
+            with open(file_name) as file:
+                result = self._http_request(
+                    method="POST",
+                    url_suffix="/api/v1/UpdateService/ImportList/Config",
+                    headers=self.headers,
+                    data={"UpdateFile": file},
+                )
+        finally:
+            shutil.rmtree(file_name, ignore_errors=True)
```

Comment:
`rmtree` is usually for directories, consider `Path(file_name).unlink()` for deleting